internal/paascharm/gogen.go

```diff
@@ -0,0 +1,78 @@
+package paascharm
+
+import (
+	"bytes"
+	_ "embed"
+	"fmt"
+	"go/format"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"text/template"
+)
+
+// CreateGoStructs reads the charmcraft.yaml file from the charmcraftDir directory
+// and outputs the Go file with the proper structs in the outputFile location,
+// using as package name packageName. It will override the outputFile file if
+// it exists and create all the parent directories if missing.
+func CreateGoStructs(charmcraftDir string, packageName string, outputFile string) (err error) {
+	yamlFile, err := os.ReadFile(path.Join(charmcraftDir, CharmcraftFileName))
+	if err != nil {
+		return fmt.Errorf("cannot read charmcraft.yaml file: %v", err)
+	}
+
+	charmcraftConfig, err := ParseCharmcraftYaml(yamlFile)
+	if err != nil {
+		return
+	}
+
+	goStructInfo, err := NewGoStructsData(packageName, charmcraftConfig)
+	if err != nil {
+		return
+	}
```

Comment:
What would these errors be? Should we give feedback to the user about what they should do in this case?

Reply:
Updated message in there. The only error for NewGoStructsData is a config option type not recognised (not one ot [these](https://juju.is/docs/sdk/charmcraft-yaml#heading--config))